### ISSUE TITLE
ScriptingCore::evalString not work

### ISSUE THREAD

**Issue by flybird** [cocos2d/cocos2d-x#16541]
- cocos2d-x version: v3.13
- devices test on:iPhone stimulate
- developing environments
  - NDK version:android-ndk-r12b
  - Xcode version:7.3.1
  - VS version:
  - brower type and version:

Steps to Reproduce:
I use latest version 3.13. I found I call ScriptingCore::evalString but It don't fire. the code always return false, the "script" is nullptr,

> JSAutoCompartment ac(cx, global);
> JS::PersistentRootedScript script(cx);
> if (script == nullptr) {
> return false;
> }

I think it should be compare with point "&script"

>  if (&script == nullptr)

**Comment by thuydx55** [comment]
I'm also getting this issue

**Comment by DavidDeSimone** [comment]
I was able to trivially reproduce this by running the line 
`CCASSERT(ScriptingCore::getInstance()->evalString("(function() { cc.log ('World Hello!'); })();"), "Testing EvalString");` in AppDelegate.cpp after ScriptingCore initialization. The reasoning seems to be that we are initializing the JS::PersistentRootedScript without an initialization value, which will mean it's operator== against nullptr will return true. Like the reporter suggests we should use the overridden operator& of the script, comparing that against nullptr. 

Also on a slightly related note, I am unsure on why we are using a JS::PersistentRootedScript here over a JS::RootedScript. The script in this use case has a definite lifetime, and is not cached or stored globally, so I see no point for using a PersistentRooted<T> here. 

I will submit a PR that fixes this issue. @pandamicro 

PR is here https://github.com/cocos2d/cocos2d-x/pull/16597

**Comment by pandamicro** [comment]
Yes, you are right, this was due to code merging, I used to new a PersistentRootedScript, and delete it at the end, but totally it's useless, let's use JS::RootedScript.

Sorry for the trouble.
